utils/converters/legacythree2gltf.js

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env node
+
+const fs = require( 'fs' );
+const path = require( 'path' );
+const Canvas = require( 'canvas' );
+const { Blob, FileReader } = require( 'vblob' );
+THREE = require( '../../build/three.js' );
+require( '../../examples/js/loaders/deprecated/LegacyJSONLoader.js' );
+require( '../../examples/js/exporters/GLTFExporter.js' );
+require( '../../examples/js/utils/BufferGeometryUtils.js' );
+
+if ( process.argv.length <= 2 ) {
+
+	console.log( `Usage: ${path.basename( __filename )} model.json [ --optimize ]` );
+	process.exit( - 1 );
+
+}
+
+var file = process.argv[ 2 ];
+var optimize = process.argv.indexOf( '--optimize' ) > 0;
+var resourceDirectory = THREE.LoaderUtils.extractUrlBase( file );
+
+//
+
+// Patch global scope to imitate browser environment.
+global.window = global;
+global.Blob = Blob;
+global.FileReader = FileReader;
+global.THREE = THREE;
+global.document = {
+	createElement: ( nodeName ) => {
+
+		if ( nodeName !== 'canvas' ) throw new Error( `Cannot create node ${nodeName}` );
+
+		const canvas = new Canvas( 256, 256 );
+		// This isn't working — currently need to avoid toBlob(), so export to embedded .gltf not .glb.
+		// canvas.toBlob = function () {
+		//   return new Blob([this.toBuffer()]);
+		// };
+		return canvas;
+
+	}
+};
+
+//
+
+// Load legacy JSON file and construct a mesh.
+
+var jsonContent = fs.readFileSync( file, 'utf8' );
+var loader = new THREE.LegacyJSONLoader();
+var { geometry, materials } = loader.parse( JSON.parse( jsonContent ), resourceDirectory );
+
+var mesh;
+var boneDefs = geometry.bones || [];
+var animations = geometry.animations || [];
+var hasVertexColors = geometry.colors.length > 0;
+
+geometry = new THREE.BufferGeometry().fromGeometry( geometry );
+
+// Remove unnecessary vertex colors and groups added during BufferGeometry conversion.
+if ( ! hasVertexColors ) geometry.removeAttribute( 'color' );
+if ( geometry.groups.length === 1 ) geometry.clearGroups();
+
+if ( ! materials ) {
+
+	materials = new THREE.MeshStandardMaterial( { color: 0x888888, roughness: 1, metalness: 0 } );
+
+}
+
+if ( optimize ) {
+
+	geometry = THREE.BufferGeometryUtils.mergeVertices( geometry );
```

Comment:
I'm not sure about this line. If the geometry has uv-coordinates, you might merge vertices with different uv-coordinates. In this case, texture coordinates will be wrong.

(BTW: This does not happen with `Geometry.mergeVertices()` because of the decoupled structure of faces and vertices.

Reply:
I think https://github.com/mrdoob/three.js/pull/14116 compares all attributes before merging vertices. But in any case, this is behind the `--optimize` flag here so it can be optionally skipped. Without that flag, the Geometry->BufferGeometry conversion was increasing the number of vertices significantly on these examples.